src/Client/Aws/SnsClient.php

```diff
@@ -70,18 +76,47 @@
  */
 class SnsClient extends AbstractAwsClient
 {
-    const RESOURCE_NAME = 'TopicArn';
+    const RESOURCE_NAME = self::SnsTopic;
+
+    private const SnsMessage = 'Message';
+    private const SnsTopic = 'TopicArn';
 
     /**
      * @param mixed $message
      *
      * @return Result
      */
-    public function send($message)
+    public function send($message): Result
+    {
+        return $this->publish([self::SnsMessage => json_encode($message),]);
+    }
+
+    /**
+     * @param MessageInterface $message
+     * @param int|null $sequence
+     * @throws \Exception
+     */
+    public function sendMessage(MessageInterface $message, ?int $sequence = null): void
+    {
+        if ($sequence) {
+            $this->sendEnvelop(new PriorityEnvelope($message, $sequence));
+            
+            return;
+        }
+        $this->sendEnvelop(new Envelope($message));
+    }
+
+    /**
+     * @param Envelope $messageEnvelop
+     * @throws \Exception
+     */
+    public function sendEnvelop(Envelope $messageEnvelop): void
     {
-        $args = ['Message' => json_encode($message)];
+        $messageEnvelop->setCreatedAt(new \DateTimeImmutable());
 
-        return $this->publish($args);
+        $this->publish([
```

Comment:
Why not returning the Result-Object here? I believe it would make things much easier to use.

Reply:
I actually didn't see the response used. I added it just in case